### PR TITLE
Adding Unit Tests

### PR DESCRIPTION
**Pull request by lidiyag**
This PR adds unit tests for kubeedge/viaduct along with the required mocks

### CODE REVIEW

File: pkg/conn/factory_test.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := NewConnection(tt.opts); reflect.TypeOf(got)!=reflect.TypeOf(tt.want) {
+				t.Errorf("NewConnection() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
```
Comment (sids-b) [inline pkg/conn/factory_test.go]: Please add blank line at the end

File: pkg/conn/factory_test.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
+package conn
+
+import (
+	"github.com/golang/mock/gomock"
+	"github.com/gorilla/websocket"
+	"github.com/kubeedge/viaduct/mocks"
+	"github.com/kubeedge/viaduct/pkg/api"
+	"reflect"
```
Comment (sids-b) [inline pkg/conn/factory_test.go]: seperate golang, kubeedge and vendor packages

File: pkg/lane/lane_test.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
+package lane
+
+import (
+	"github.com/golang/mock/gomock"
+	"github.com/kubeedge/viaduct/mocks"
+	"github.com/kubeedge/viaduct/pkg/api"
+	"github.com/gorilla/websocket"
+	"reflect"
```
Comment (sids-b) [inline pkg/lane/lane_test.go]: seperate golang, kubeedge and vendor packages

File: pkg/lane/quic_test.go [hunk 4]
```diff
@@ -0,0 +1,182 @@
+package lane
+
+import (
+	"errors"
+	"github.com/golang/mock/gomock"
+	"github.com/kubeedge/beehive/pkg/core/model"
```
Comment (sids-b) [inline pkg/lane/quic_test.go:12]: seperate golang, kubeedge and vendor packages

File: pkg/lane/quic_test.go [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+			want:&QuicLane{stream:mockStream},
+		},
+		{
+			name:"NewQuickLaneTest-EmptyObject",
+			van:"",
+			want:nil,
+		},
+	}
+	for _, tt := range tests {
+
```
Comment (sids-b) [inline pkg/lane/quic_test.go]: remove extra blank line

File: pkg/lane/quic_test.go [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+			msg:&model.Message{Content:"message"},
+			wantErr:nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			l := &QuicLane{
+				stream:        tt.stream,
+			}
+			if ( tt.name == "Test-FailureCase") {
```
Comment (sids-b) [inline pkg/lane/quic_test.go]: the main idea of table driven tests is easy addition of test. We have if else condition based on name of the test which should be unique for all tests, if we want to add 3rd we may require another else if. Please refactor.

File: pkg/lane/quic_test.go [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+			if got := NewQuicLane(tt.van); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("NewQuicLane() = %v, want %v", got, tt.want)
+			}
+
+		})
+	}
+}
+
+// TestQuicLane_ReadMessage is function to test ReadMessage().
+func TestQuicLane_ReadMessage(t *testing.T) {
```
Comment (sids-b) [inline pkg/lane/quic_test.go]: camel case please.
Applicable to similar cases below

File: pkg/lane/quic_test.go [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+			}
+			if err := l.ReadMessage(tt.msg); (err != tt.wantErr) {
+				t.Errorf("QuicLane.ReadMessage() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+// TestQuicLane_WriteMessage is function to test WriteMessage().
+func TestQuicLane_WriteMessage(t *testing.T) {
```
Comment (sids-b) [inline pkg/lane/quic_test.go]: camel case please

File: pkg/packer/reader_test.go [hunk 9]
```diff
@@ -0,0 +1,95 @@
+package packer
+
+import (
+	"errors"
+	"github.com/golang/mock/gomock"
+	"github.com/kubeedge/beehive/pkg/core/model"
+	"github.com/kubeedge/viaduct/mocks"
+	"github.com/kubeedge/viaduct/pkg/translator"
```
Comment (sids-b) [inline pkg/packer/reader_test.go:13]: seperate golang, kubeedge and vendor packages

File: pkg/lane/quic_test.go [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+func TestQuicLane_SetWriteDeadline(t *testing.T) {
+	tests := []struct {
+		name    string
+		writeDeadline time.Time
+		stream        quic.Stream
+		t time.Time
+		wantErr bool
+	}{
+		{
+			name:"Test1",
```
Comment (sids-b) [inline pkg/lane/quic_test.go]: Please give a name that desribes the test in short.
Applicable to similar places.

File: pkg/packer/reader_test.go [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+// TestRead is function to test Read().
+func TestRead(t *testing.T) {
+	initMocks(t)
+	var ioreader io.Reader
+	var msg  = model.Message{Content:"message"}
+	bytesMsg, _ :=translator.NewTran().Encode(&msg)
+	header := PackageHeader{Version:0011,PayloadLen:(uint32(len(bytesMsg)))}
+	headerBuffer := make([]byte, 0)
+	header.Pack(&headerBuffer)
+	err1:=errors.New("Error")
```
Comment (sids-b) [inline pkg/packer/reader_test.go]: Lets not suffix names with numbers. Please change it to something that says why this error is used.

**Review by sids-b — CHANGES_REQUESTED** [review]
Please gofmt all files.

File: mocks/cover.out [hunk 12]
```diff
@@ -0,0 +1 @@
+mode: set
```
Comment (sids-b) [inline mocks/cover.out]: Why is this file used ??

Comment (lidiyag) [reply mocks/cover.out]: Pushed by mistake...Updated.

**Review by sids-b — APPROVED** [review]
LGTM

**Pull request merged**